src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationevents/services/HmppsDomainEventService.kt

```diff
@@ -46,10 +47,11 @@ class HmppsDomainEventService(
       return crn
     }
     val nomsNumber = hmppsEvent.personReference.findNomsIdentifier()
-    if (nomsNumber != null) {
-      val identifier = probationIntegrationApiGateway.getPersonIdentifier(nomsNumber)
-      return identifier?.crn
+
+    nomsNumber?.let {
+      return probationIntegrationApiGateway.getPersonIdentifier(nomsNumber)?.crn ?: throw NotFoundException("Person not found nomsNumber $nomsNumber")
```

Comment:
Is this intentional that we're throwing an error here now?  We were returning null before.

Reply:
return null would result in error 'CRN could not be found in registration event message'.  this is to distinguish between crn or noms number not present in payload in person identifier (unlikely), compared to call to backend and noms ref does not exist.